### ISSUE TITLE
Prevent reload on a purged environment

### ISSUE THREAD

**Issue by amercader** [boxkite/datacats#51]
So I wanted to reset an environment:

`datacats purge spatial22`

Still not being familiar with datacats I run `datacats reload spatial22`. It fails with:

```
(datacats)➜  spatial22  datacats reload                                   
Traceback (most recent call last):
  File "/home/adria/dev/pyenvs/datacats/bin/datacats", line 9, in <module>
    load_entry_point('datacats==0.7', 'console_scripts', 'datacats')()
  File "/home/adria/dev/pyenvs/datacats/local/lib/python2.7/site-packages/datacats/cli/main.py", line 115, in main
    return command_fn(project, opts)
  File "/home/adria/dev/pyenvs/datacats/local/lib/python2.7/site-packages/datacats/cli/manage.py", line 77, in reload_
    project.start_web(opts['--production'])
  File "/home/adria/dev/pyenvs/datacats/local/lib/python2.7/site-packages/datacats/project.py", line 454, in start_web
    self._create_run_ini(port, production)
  File "/home/adria/dev/pyenvs/datacats/local/lib/python2.7/site-packages/datacats/project.py", line 493, in _create_run_ini
    makedirs(self.datadir + '/run')  # upgrade old datadir
  File "/home/adria/dev/pyenvs/datacats/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 13] Permission denied: '/home/adria/.datacats/spatial22/run'
```
The directory looks like:

```
(datacats)➜  spatial22  ll /home/adria/.datacats/
total 24K
drwx------ 9 adria adria 4.0K Mar 22 21:14 plain
drwx------ 9 adria adria 4.0K Mar 23 21:57 plain2
drwx------ 9 adria adria 4.0K Mar 22 18:54 plain23
drwxr-xr-x 4 root  root  4.0K Mar 25 18:21 spatial22
drwx------ 9 adria adria 4.0K Mar 25 18:16 spatial23
drwx------ 7 adria adria 4.0K Mar 25 18:19 spatial24
``` 

I looks like I should have done `datacats init` instead, but maybe the `reload` command should have stopped before.

Also not sure where this escalation of privileges came from, as I was running the command with my own user.

**Comment by wardi** [comment]
@amercader thanks! would you try the purge command from the current master version? I've fixed some of the places it can fail when cleaning up datadirs that were only half-created (because of the earlier failure you ran into with create)

**Comment by wardi** [comment]
docker creates volume dirs as root by default. We're trying to reset all that in python code but we haven't taken care of all the corner cases, like when create or init fails.

**Comment by wardi** [comment]
fixed by #55
